Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/recovery/core/RecoveryCertificate.kt

```diff
@@ -7,7 +7,10 @@ import org.joda.time.LocalDate
 
 interface RecoveryCertificate : CwaCovidCertificate {
     override val containerId: RecoveryCertificateContainerId
-    val testedPositiveOn: LocalDate
+    val testedPositiveOnFormatted: String
+    val validFromFormatted: String
+    val validUntilFormatted: String
```

Comment:
is it a general requirement that we need to add here and we should use ? can not we just format it based on the use case ?

Reply:
Yes, tech spec defines how they must be formatted. To prevent mix up in formatting (which has been the case so far), I decided to expose the formatted String only. The local date parsed from that formatted String is only exposed in case there are calculations with the date. This also makes it easier in case further changes become necessary.

Reply:
But is TestCertificate for example  has different formatting in case when it is the highest prio certificate (under qrcode)
and  when it is in the list below the qr code both are in the person details screen ,can we still access the `LocaleDate ` ?

Reply:
For formatting on screens that do not need to follow the tech spec, local date is exposed

Reply:
👍